src/array/linestring/mutable.rs

```diff
@@ -155,6 +174,32 @@ impl<O: OffsetSizeTrait> MutableLineStringArray<O> {
     pub fn into_array_ref(self) -> Arc<dyn Array> {
         Arc::new(self.into_arrow())
     }
+
+    pub fn from_line_strings(
+        geoms: &[impl LineStringTrait<T = f64>],
+        coord_type: Option<CoordType>,
+    ) -> Self {
+        let (coord_capacity, geom_capacity) = first_pass(geoms.iter().map(Some));
+        second_pass(
+            geoms.iter().map(Some),
+            coord_capacity,
+            geom_capacity,
+            coord_type.unwrap_or_default(),
+        )
+    }
```

Comment:
Hmm maybe a clean way to make this a public API would be to rename:

- first_pass -> `with_capacities_from_iter`
- second_pass -> `extend_from_iter`